.github/workflows/android_deploy_production_to_playstore.yml

```diff
@@ -0,0 +1,61 @@
+name: Android - Deploy Production build to Play Store
+
+on:
+  push:
+    branches:
+      - main
+      - feature/61-setup-cd-to-build-and-deploy-android-app-to-playstore
+
+jobs:
+  build_and_deploy_android:
+    name: Build & Deploy Android
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v3
+
+      - name: Setup Python
+        uses: actions/setup-python@v2
+        with:
+          python-version: '3.9'
+
+      - name: Generate new project
+        run: make run PACKAGE_NAME=co.nimblehq.flutter.template PROJECT_NAME=flutter_templates APP_NAME="Flutter Templates"
+
+      - name: Setup Java JDK
+        uses: actions/setup-java@v3
+        with:
+          distribution: 'adopt'
+          java-version: '11'
+
+      - name: Setup Flutter environment
+        uses: subosito/flutter-action@v2
+        with:
+          channel: 'stable'
+          flutter-version: '3.3.10'
+
+      - name: Get flutter dependencies
+        run: flutter pub get
+
+      - name: Run code generator.
+        run: flutter packages pub run build_runner build --delete-conflicting-outputs
+
+      - name: Setup .env
+        env:
+          ENV_PRODUCTION: ${{ secrets.ENV_PRODUCTION }}
+        run: |
+          echo $ENV_PRODUCTION > .env
+
+      - name: Build Production App Bundle
+        run: flutter build appbundle --flavor production --release --build-number $GITHUB_RUN_NUMBER
+
+      # TODO We will enable this step later when having a shared Service Account on our organization
+      # Basically, this step will work properly after providing a valid Service Account.
+      # - name: Upload Android Production Release bundle to Play Store
+      #   uses: r0adkll/upload-google-play@v1
+      #   with:
+      #     serviceAccountJson: ${{ secrets.GOOGLE_PLAY_SERVICE_ACCOUNT_JSON }}
+      #     packageName: co.nimblehq.flutter.template
+      #     releaseFile: build/app/outputs/bundle/productionRelease/app-production-release.aab
+      #     track: internal
+      #     userFraction: 1.0
+      #     whatsNewDirectory: .github/workflows/whatsnew
```

Comment:
FYI: I created a separate ticket for a later update https://github.com/nimblehq/flutter_templates/issues/168.